fsm/fsm.go

```diff
@@ -81,18 +81,23 @@ type Observer interface {
 	Notify(Notification)
```

Comment:
nit: could you create a separate PR for `fsm/fsm.go` and `fsm/observer.go` changes?